cmd/core/pfcp_session_handlers.go

```diff
@@ -371,8 +395,16 @@ func HandlePfcpSessionModificationRequest(conn *PfcpConnection, msg message.Mess
 
 			spdrInfo := SPDRInfo{}
 			if err := extractPDR(pdr, session, &spdrInfo); err == nil {
+				if HasSdf(pdr) {
+					pdrId, spdrInfo, err = session.GetPdrWithAdditionalRules(spdrInfo)
```

Comment:
If I understand correctly the code, you try to find PDR in session and then add SDF filter to that PDR. But for userspace session representation we don't need to piggyback SDF to root PDR. For the userspace we can store them as different PDR each with own ID(strictly as stated in PFCP command).

SDF piggybacking is needed only when we write ebpf map structures.

Reply:
Done?